### DISCUSSION TITLE
Take Control not connecting

### DISCUSSION THREAD

**Discussion by ccasalicchio** [amidaware/tacticalrmm#1077] [General]
**Server Info:**
 - OS: Ubuntu 20.04
 - Browser: chrome
 - RMM Version: v0.12.4

**Installation Method:**
  - [X] Standard
  - [ ] Docker

**Agent Info:**
- Agent version: Agent v2.0.2
- Agent OS: Windows 10/11

**Describe the bug**
The agent works just fine, but when I try to run the "Take Control" it just never connects. I believe this could be a DNS/Reverse proxy issue, as we are running the RMM server behind an Nginx server.

NGINX config:
```
server { 
 listen 80; 
 server_name 
remotermm.site.com api.site.com rmm.site.com;
 location / { 
 proxy_pass http://xxx.xxx.xxx.xxx:80/;  
 proxy_http_version 1.1; 
 # Inform MeshCentral about the real host, port and protocol 
 proxy_set_header X-Forwarded-Host $host:$server_port; 
 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 
 proxy_set_header X-Forwarded-Proto $scheme; 
 } 
 } 
 # HTTPS server. In this example, we use a wildcard as server name. 
 server { 
 listen 443 ssl; 
 server_name 
remotermm.site.com api.site.com rmm.site.com;
  
 # MeshCentral uses long standing web socket connections, set longer timeouts. 
 proxy_send_timeout 330s; 
 proxy_read_timeout 330s; 
 # We can use the MeshCentral generated certificate & key 
 ssl_certificate    /www/server/panel/vhost/cert/remotermm.site.com/fullchain.pem;
    ssl_certificate_key    /www/server/panel/vhost/cert/remotermm.site.com/privkey.pem;

 ssl_session_cache shared:WEBSSL:10m; 
 ssl_ciphers HIGH:!aNULL:!MD5; 
 ssl_prefer_server_ciphers on; 
 location / { 
 proxy_pass https://xxx.xxx.xxx.xxx:443/;  
 proxy_http_version 1.1; 
 # Allows websockets over HTTPS. 
 proxy_set_header Upgrade $http_upgrade; 
 proxy_set_header Connection "upgrade"; 
 proxy_set_header Host $host; 
 # Inform MeshCentral about the real host, port and protocol 
 proxy_set_header X-Forwarded-Host $host:$server_port; 
 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 
 proxy_set_header X-Forwarded-Proto $scheme; 
 } 
 }

```
**To Reproduce**
1. Login
2. Navigate to the computer
3. Click "Take Control"

Pop-up opens but never connects
![image](https://user-images.githubusercontent.com/2466879/164310743-ad5f7e8d-1486-43bd-8c41-33a50e5e7492.png)

**Expected behavior**
Should open the remote screen

**Screenshots**
![image](https://user-images.githubusercontent.com/2466879/164310799-298b25c9-2255-4ceb-9067-b0b50ae27088.png)

**Additional context**
This is a new installation, got the agents to work, but mesh is not connecting apparently

**Comment by dinger1986** [comment]
Yes absolutely is a proxy issue.

Have you checked unsupported on the docs?

This isn't an issue with TRMM rather it being used unsupported.

Our discord group is extremely active and plenty of users who have implemented reverse proxys

**Reply by ccasalicchio** [reply]
Thank you, I'll check it out

**Comment by wh1te909** [comment]
the install script already configures nginx as a reverse proxy. if you already have an nginx server, then you're basically running nginx in front of nginx which doesn't make sense and probably causing issues. you should disable nginx on the rmm server and just use your own nginx. take a look at https://docs.tacticalrmm.com/unsupported_proxies/ for some pointers

**Reply by ccasalicchio** [reply]
Thanks
